libtenzir/include/tenzir/optional.hpp

```diff
@@ -20,3 +22,21 @@ std::optional<T> to_optional(const T* ptr) {
 }
 
 } // namespace tenzir
+
+namespace fmt {
+template <class T>
+struct formatter<std::optional<T>> {
+  template <class ParseContext>
+  constexpr auto parse(ParseContext& ctx) {
+    return ctx.begin();
+  }
+
+  template <class FormatContext>
+  auto format(const std::optional<T>& value, FormatContext& ctx) const {
+    if (!value) {
+      return fmt::format_to(ctx.out(), "none");
+    }
+    return fmt::format_to(ctx.out(), "{}", *value);
+  }
+};
+} // namespace fmt
```

Comment:
I believe this can just be dropped and we bump the minimum fmt to 10.

Reply:
Unfortunately I have to retract that. All reasonable released distros are (debian-12, ubuntu-24.04) are still on fmt_9, and `debian-trixie` is a moving target that can fail in CI at any time.